src/routes/v0/tag/global.rs

```diff
@@ -75,53 +99,77 @@ pub async fn tag_taggers_handler(
 ) -> Result<Json<Vec<String>>> {
     info!(
-        "GET {TAG_TAGGERS_ROUTE} label:{}, reach:{:?}",
-        label, query.reach
+        "GET {TAG_TAGGERS_ROUTE} label:{}, query: {:?}",
+        label, query
     );
 
-    let reach = query.reach;
+    let skip = query.pagination.skip.unwrap_or(0);
+    let limit = query.pagination.limit.unwrap_or(20);
 
-    match TagGlobal::get_tag_taggers(label.clone(), reach).await {
+    match TagGlobal::get_tag_taggers(label.clone(), query.user_id, query.reach, skip, limit).await {
         Ok(Some(post)) => Ok(Json(post)),
         Ok(None) => Err(Error::TagsNotFound { reach: label }),
         Err(source) => Err(Error::InternalServerError { source }),
     }
 }
 
 #[derive(Deserialize)]
-pub struct TagsByReachQuery {
+pub struct TagsByReachPath {
     user_id: String,
     reach: Option<TagStreamReach>,
 }
 
 #[utoipa::path(
     get,
-    path = TAG_REACH_ROUTE,
-    tag = "Global Tags by reach",
+    path = HOT_TAGS_BY_REACH_ROUTE,
+    tag = "Hot Tags by reach",
     params(
         ("user_id" = String, Path, description = "User Pubky ID"),
-        ("reach" = TagStreamReach, Path, description = "Reach type: Follower | Following | Friends")
+        ("reach" = TagStreamReach, Path, description = "Reach type: Follower | Following | Friends"),
+        ("taggers_limit" = Option<usize>, Query, description = "Retrieve N user_id for each tag"),
+        ("skip" = Option<usize>, Query, description = "Skip N tags"),
+        ("limit" = Option<usize>, Query, description = "Retrieve N tag"),
+        ("from" = Option<i64>, Query, description = "Retrieve hot tags from this timestamp"),
+        ("to" = Option<i64>, Query, description = "Retrieve hot tags up to this timestamp"),
+        ("tagged_type" = Option<TaggedType>, Query, description = "Retrieve hot tags by the type of entities tagged with it"),
     ),
     responses(
         (status = 200, description = "Retrieve tags by reach cluster", body = Vec<HotTag>),
         (status = 404, description = "Hot tags not found"),
         (status = 500, description = "Internal server error")
     )
 )]
-pub async fn tags_by_reach_handler(Path(path): Path<TagsByReachQuery>) -> Result<Json<HotTags>> {
+pub async fn tags_by_reach_handler(
+    Path(path): Path<TagsByReachPath>,
+    Query(query): Query<HotTagsQuery>,
+) -> Result<Json<HotTags>> {
     info!(
-        "GET {TAG_REACH_ROUTE}: {:?}, {:?}",
-        path.user_id, path.reach
+        "GET {HOT_TAGS_BY_REACH_ROUTE} user_id: {:?}, reach: {:?}, query: {:?}",
+        path.user_id, path.reach, query
     );
 
     let reach = path.reach.unwrap_or(TagStreamReach::Following);
     let user_id = path.user_id;
 
-    match HotTags::get_stream_tags_by_reach(user_id, reach.clone()).await {
+    let skip = query.pagination.skip.unwrap_or(0);
+    let limit = query.pagination.limit.unwrap_or(40);
+    let taggers_limit = query.taggers_limit.unwrap_or(20);
+    let from = query.from.unwrap_or(0);
```

Comment:
Timeframe pagination with `from` and `to` should not be exposed to the end users. There should be 3 pre-defined options. All, 24h and 1 month.

For reach all -> we precompute the sorted set every 24h. We can do that with a cronjob or by having a TTL of 24 hours and a fallback (on-cache-miss) to graph (re-populates the cache).